trinity/utils/chains.py

```diff
@@ -60,55 +62,61 @@ def get_data_dir_for_network_id(network_id: int) -> str:
 NODEKEY_FILENAME = 'nodekey'
 
 
-def get_nodekey_path(data_dir: str) -> str:
+def get_nodekey_path(data_dir: PurePath) -> PurePath:
     """
     Returns the path to the private key used for devp2p connections.
     """
-    return os.environ.get(
+    return PurePath(os.environ.get(
         'TRINITY_NODEKEY',
-        os.path.join(data_dir, NODEKEY_FILENAME),
-    )
+        str(data_dir / NODEKEY_FILENAME),
+    ))
 
 
 DATABASE_SOCKET_FILENAME = 'db.ipc'
 
 
-def get_database_socket_path(data_dir: str) -> str:
+def get_database_socket_path(data_dir: PurePath) -> str:
     """
     Returns the path to the private key used for devp2p connections.
+
+    We're still returning 'str' here on ipc-related path because an issue with
+    multi-processing not being able to interpret 'PurePath' objects correctly.
     """
     return os.environ.get(
         'TRINITY_DATABASE_IPC',
-        os.path.join(data_dir, DATABASE_SOCKET_FILENAME),
+        os.path.join(str(data_dir), DATABASE_SOCKET_FILENAME),
```

Comment:
I think the approach you did above method is more of a standard `PurePath` approach (but I mostly care about internal consistency).

```py
str(data_dir / DATABASE_SOCKET_FILENAME)
```